### ISSUE TITLE
Recommendations for V3 can

### ISSUE THREAD

**Issue by EliavRevah** [WhyAsh5114/MyFit#6]
1.The option to choose different sets for exercises: myoreps,Myo match, down sets, drop sets etc...

2. Exercise database so the users can select the exercise they wish to add 

3.Option to set exercises to multiple muscle groups for example instead of having  chest press set to only chest you could set it to chest triceps and front delts

4. A counter of partial reps per set

**Comment by WhyAsh5114** [comment]
I'm not sure how to implement the exercise database yet, since not all users will abide by the conventional way to do a exercise. For example: a pull up could be used for focusing on the back/biceps.

A possible implementation would be to separate user exercises and common exercises.
A search button when adding exercises would show both user and common exercises, selecting an exercise would then populate the fields accordingly for quick addition. How does that sound?

**Comment by EliavRevah** [comment]
Sounds great!

**Comment by Leech025** [comment]
> I'm not sure how to implement the exercise database yet, since not all users will abide by the conventional way to do a exercise. For example: a pull up could be used for focusing on the back/biceps.
> 
> A possible implementation would be to separate user exercises and common exercises. A search button when adding exercises would show both user and common exercises, selecting an exercise would then populate the fields accordingly for quick addition. How does that sound?

Just my 2 cents on the topic: I like the idea of common exercise database and the ability to add separate user exercises, and in addition perhaps there could be a streamlined way to modify an exercise (either default 'common exercises' or 'user-defined exercises' and save a derivative? Use case 1 = Sometimes I do bent rows pulling bar to chest, but sometimes I do bar to stomach and it would be nice to keep them separate for volume management purposes; Use case 2 = some sets I program for lighter squeeze sets @ 20-30 reps but most of the time do medium weight for 10-20 of the same exercise, and imputing every variation in form and rep range is a bit cumbersome.

**Comment by WhyAsh5114** [comment]
Yeah, using the common exercises database would just populate the fields when adding an exercise with sensible defaults, or your own user exercise's properties, it won't directly add it before prompting for any changes. 
And you can simply change whatever u wish to, simplifying exercise addition.

**Comment by iamjexcdevvv** [comment]
Hello Dev I just found out your app and it really catch my attention first of all the app seems really amazing. I'm here just to recommends something from my mind which is the feature I used from other fitness app idk if RP app has this feature as well.

1. Rest Timer for every set (it depends on you on how you implement this) and you can also adds an alarm if the timer goes off.

2. Also the offline support it will be great to have this feature in this app

I will keep on recommending features to help you 💪💪💪

**Comment by WhyAsh5114** [comment]
hi @iamjexcdevvv, these are good suggestions, thanks a lot! Offline support is definitely gonna be there in v3, for the rest times, maybe I'll just add a UI thing for it, but I don't think it'll be worth saving in the database as they aren't that important, but a neat feature for convenience.

**Comment by iamjexcdevvv** [comment]
> hi @iamjexcdevvv, these are good suggestions, thanks a lot! Offline support is definitely gonna be there in v3, for the rest times, maybe I'll just add a UI thing for it, but I don't think it'll be worth saving in the database as they aren't that important, but a neat feature for convenience.

Hi @WhyAsh5114 not to give you a pressure to release the V3 immediately but I just want to know if there is a progress now? it's a month ago since the last update from your repository

**Comment by WhyAsh5114** [comment]
Hey @iamjexcdevvv, I have my final semester exams right now so I am not able to contribute as much as I used to, but they'll be over by the end of June. I do contribute in the repository but not in the `main` branch, most of my work is currently in the `v3-svelte5` branch
